src/main/java/com/smartbear/readyapi/client/execution/RecipeExecutor.java

```diff
@@ -108,17 +108,23 @@ public HarLogRoot getTransactionLog(final Execution execution, String transactio
         return executions;
     }
 
-    private Execution doExecuteTestCase(TestCase testCase, boolean async) {
+    private Execution doExecuteTestCase(TestCase testCase, boolean async) throws ApiException {
         try {
             ProjectResultReport projectResultReport = apiStub.postTestRecipe(testCase, async, authentication);
             cancelExecutionAndThrowExceptionIfPendingDueToMissingClientCertificate(projectResultReport, testCase);
             return new Execution(apiStub, authentication, projectResultReport);
         } catch (ApiException e) {
             invokeListeners(e);
             logger.debug("An error occurred when sending test recipe to server. Details: " + e.toString());
+            if (!async) {
+                throw e;
+            }
         } catch (Exception e) {
             invokeListeners(e);
             logger.debug("An error occurred when sending test recipe to server", e);
+            if (!async) {
```

Comment:
my bad... I think should still propagate errors during async submits as exceptions...